homeassistant/components/mqtt/climate.py

```diff
@@ -798,6 +883,26 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
             return
         self.async_write_ha_state()
 
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+
+        if not self.supported_features & ClimateEntityFeature.TARGET_HUMIDITY:
+            _LOGGER.error(
+                "Attempted to set target humidity with no %s configured",
+                CONF_HUMIDITY_COMMAND_TOPIC,
+            )
+            return
```

Comment:
I don't think this is needed, the base entity already checks if setting humidity is supported: https://github.com/home-assistant/core/blob/db3e57542558cf06c2ec9c7f009626466569b0d2/homeassistant/components/climate/__init__.py#L170-L176

Reply:
Added a sugestion to remove the check.